src/users/user.php

```diff
@@ -28,29 +28,22 @@ class UsersApiResourceUser extends ApiResource
 	 */
 	public function post()
 	{
-		$app = JFactory::getApplication();
-		$userIdentifier = $app->input->get('id', 0, 'String');
-		$formData = $app->input->getArray();
-		$params = JComponentHelper::getParams("com_users");
-		$response = new stdClass;
-
-		$xidentifier = $app->input->server->get('HTTP_X_IDENTIFIER');
-		$fidentifier = $app->input->server->get('HTTP_FORCECREATE');
-
-		if ($formData['username'] == '' || $formData['name'] == '' || $formData['email'] == '')
-		{
-			ApiError::raiseError(400, JText::_('PLG_API_USERS_REQUIRED_DATA_EMPTY_MESSAGE'));
-
-			return;
-		}
+		$app            = JFactory::getApplication();
+		$params         = JComponentHelper::getParams("com_users");
+		$formData       = $app->input->getArray();
+		$userIdentifier = $app->input->get('id', 0, 'string');
+		$xIdentifier    = $app->input->server->get('HTTP_X_IDENTIFIER');
+		$fIdentifier    = $app->input->server->get('HTTP_FORCECREATE');
 
 		// Get current logged in user.
-		$my = JFactory::getUser();
+		$me = JFactory::getUser();
```

Comment:
The user object is already available in `$this->plugin->get('user')` can you confirm it is ok to use either ?